src/matter/cluster/OnOffCluster.ts

```diff
@@ -4,18 +4,109 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { BooleanT } from "../../codec/TlvObjectCodec";
-import { Attribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { BooleanT, Field, ObjectT, StringT, Template, UnsignedIntT, UnsignedLongT } from "../../codec/TlvObjectCodec";
+import { Attribute, OptionalAttribute, OptionalWritableAttribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { TlvType } from "../../codec/TlvCodec";
 
+export const enum OnOffStartUpOnOff {
+    Off = 0,
+    On = 1,
+    Toggle = 2,
+}
+const OnOffStartUpOnOffT = { tlvType: TlvType.UnsignedInt } as Template<OnOffStartUpOnOff>;
+
+export const enum OnOffEffectIdentifier {
+    DelayedAllOff = 0,
+    DyingLight = 1,
+}
+const OnOffEffectIdentifierT = { tlvType: TlvType.UnsignedInt } as Template<OnOffEffectIdentifier>;
+
+export const enum OnOffDelayedAllOffEffectVariant {
+    FadeToOffIn_0p8Seconds = 0,
+    NoFade = 1,
+    "50PercentDimDownIn_0p8SecondsThenFadeToOffIn_12Seconds" = 2,
+}
+const OnOffDelayedAllOffEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDelayedAllOffEffectVariant>;
+
+export const enum OnOffDyingLightEffectVariant {
+    "20PercentDimUpIn_0p5SecondsThenFadeToOffIn_1Second" = 0,
+}
+const OnOffDyingLightEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDyingLightEffectVariant>;
+
+/*
+  <bitmap name="OnOffControl" type="BITMAP8">
```

Comment:
I woulkd need a type definition for the "bit map" fields. or do we do ENUMS and users need to use enum.A & enum.B or such to set multiple bits and use a number? or what do we do?

Reply:
awesome! I add it as TODO in these places and we adjust after merge